File: apps/frontend/src/api/reports.ts
import { apiFetch } from "./client";

export type SummaryItem = {
  work_id: string;
  title: string;
  sprint_name: string;
  total_seconds: number;
  total_earned_cents: number;
  currency: string;
};

export type SummaryResponse = {
  from: string;
  to: string;
  total_seconds: number;
  total_earned_cents: number;
  currency: string;
  by_work: SummaryItem[];
};

export async function getSummary(dateFrom: string, dateTo: string) {
  const q = new URLSearchParams({ date_from: dateFrom, date_to: dateTo });
  return apiFetch<SummaryResponse>(`/reports/summary?${q.toString()}`);
}

File: apps/backend/app/schemas/reports.py
from pydantic import BaseModel

class WorkSummaryItem(BaseModel):
    work_id: str
    title: str
    sprint_name: str
    total_seconds: int
    total_earned_cents: int
    currency: str

class SummaryResponse(BaseModel):
    from_: str
    to: str
    total_seconds: int
    total_earned_cents: int
    currency: str
    by_work: list[WorkSummaryItem]
